packages/web3-eth-ens/src/lib/ResolverMethodHandler.js

```diff
@@ -69,9 +84,11 @@ ResolverMethodHandler.prototype.call = function (callback) {
     var self = this;
     var promiEvent = new PromiEvent();
     var preparedArguments = this.parent.prepareArguments(this.ensName, this.methodArguments);
+    var outputFormatter = this.outputFormatter || null;
 
-    this.parent.registry.getResolver(this.ensName).then(function (resolver) {
-        self.parent.handleCall(promiEvent, resolver.methods[self.methodName], preparedArguments, callback);
+    this.parent.registry.getResolver(this.ensName).then(async function (resolver) {
+        await self.parent.checkInterfaceSupport(resolver, self.methodName);
```

Comment:
I remember a discussion I had once with @cgewecke about additional network requests and their impact. Shouldn't this ``checkInterface`` call not also be optional with a ``strictEns`` flag or so? Otherwise, the arguments in the discussion we had are false:)

Reply:
The origin of this design is the discussion beginning in [#3392 comment][1]. In this case there are two widely deployed resolvers with different APIs, so there's a strong case for validating the request before-hand. 

I think you're alluding here to a [debate in #3428][1] about whether or not to gate revert-with-reason behind an option flag. The ENS module is not subject to the similar considerations AFAIK, personally don't think this will impact people in the same way 🤷  

[1]: https://github.com/ethereum/web3.js/pull/3248#pullrequestreview-324469285




[1]: https://github.com/ethereum/web3.js/issues/3392#issuecomment-591918724